File: Day 2 - Java/Code/src/Employee.java

public abstract class Employee {
	
	private static int starting;
	
	static {	//static initializer block;
		System.out.println("Employee class loaded...");
		starting = 100;
	}
	private int empNo;
	private String empName;
	private double salary;
	
	
	public Employee(String empName, double salary) {
		this.empNo = ++starting;
		this.empName = empName;
		this.salary = salary;
	}
	
	public void payslip() {
		System.out.println("\nEmp No: " + empNo);
		System.out.println("Name: " + empName);
		System.out.println("Salary: " + salary);
	}

	public double getSalary() {
		return salary;
	}
	
	public String getEmpName() {
		return this.empName;
	}


}

File: Day 19 - Docker, JS/Frontend/Node/employee.js
const Person = require("./person").Person

class Employee extends Person{
    constructor(name, age, salary){
        super(name, age);
        this.salary = salary;
    }

    payslip = () => {
        this.print();
        console.log("Salary: " + this.salary);
    }
}

let e = new Employee("Polo", 25, "45000");
e.payslip();